esphome/components/modbus_controller/__init__.py

```diff
@@ -129,6 +130,7 @@
             cv.Optional(
                 CONF_COMMAND_THROTTLE, default="0ms"
             ): cv.positive_time_period_milliseconds,
+            cv.Optional(CONF_MAX_CMD_RETRIES, default=5): cv.positive_int,
```

Comment:
```suggestion
            cv.Optional(CONF_MAX_CMD_RETRIES, default=5): cv.int_range(min=1),
```
A value of zero would mean the command is never sent. I'm assume that's not meant to be an option.

Reply:
This is a little confusing I admit, but if this is a retry, I could have this as zero. Because if the command is sent and receives a response, there is no retry per se. What are your thoughts about it?

Reply:
I'll keep the way it is, and make it clear in the documentation that this number includes the initial sent. If you agree, of course.

Reply:
Sure, though I can't imagine why anyone would want to specify this and never have it sent. An alternative would be to allow it to be zero, but add 1 when setting to allow for the initial send. That would make sense in that the "retries" value is just that.

Reply:
The issue is just the nomenclature; if we say this is the retry count, the initial transmission is not a retry, so it should not be counted. I'll add 1 and make sure the documentation makes it clear.